playwright-e2e/dsm/component/tabs/sequencing-order-tab.ts

```diff
@@ -0,0 +1,101 @@
+import { Locator, Page, expect } from '@playwright/test';
+import { getDate } from 'utils/date-utils';
+
+export default class SequeuncingOrderTab {
+  constructor(private readonly page: Page) {
+  }
+
+  /* Actions */
+
+  public async waitForReady(): Promise<void> {
+    const sequencingTab = this.page.locator(`//tab[@heading='Sequencing Order']`);
+    await sequencingTab.scrollIntoViewIfNeeded();
+    await expect(sequencingTab, 'The Sequencing Tab is not currently active').toHaveClass(/active/);
+  }
+
+  public async placeOrder(): Promise<void> {
+    const placeOrderButton = this.page.getByRole('button', { name: 'Place order' });
+    await placeOrderButton.scrollIntoViewIfNeeded();
+    await placeOrderButton.click();
+  }
+
+  public async assertNoSamplesCanBeSelected(): Promise<void> {
+    const allAvailableSamples = await this.getAllPossibleSamples();
+    for (const sample of allAvailableSamples) {
+      const sampleCheckbox = this.getCheckboxOfSample(sample);
+      await expect(sampleCheckbox).not.toBeVisible();
+    }
+  }
+
+  public async assertPlaceOrderButtonNotVisible(): Promise<void> {
+    const placeOrderButton = this.getPlaceOrderButton();
+    await expect(placeOrderButton).not.toBeVisible();
+  }
+
+  public async fillAvailableCollectionDateFields(opts: {canPlaceClinicalOrder?: boolean}): Promise<void> {
+    const { canPlaceClinicalOrder = true } = opts;
+    const collectionDateIndex = canPlaceClinicalOrder ? 5 : 4; //Index depending on whether test user has permission to place clinical order
+    const allAvailableSamples = await this.getAllPossibleSamples();
+    for (const sample of allAvailableSamples) {
+      const collectionDateSection = sample.locator(`//td[${collectionDateIndex}]`); //Get Collection Date section
+      if (await this.isInteractiveDateField(collectionDateSection)) {
+        const collectionDateField = collectionDateSection.locator(`//input[@data-placeholder='mm/dd/yyyy']`); //Get date field
+        const today = getDate();
+        await collectionDateField.fill(today);
+      }
+    }
+  }
+
+  private async isInteractiveDateField(dateField: Locator): Promise<boolean> {
+    const fieldInput = await dateField.innerText();
+    if (fieldInput === `mm/dd/yyyy\n  Today`) {
+      return true;
+    }
+    return false;
+  }
+
+  /* Locators */
+  public async getAllPossibleSamples(): Promise<Locator[]> {
+    return this.page.locator(`//app-sequencing-order//tr[contains(.,'Normal') or contains(.,'Tumor')]`).all();
+  }
+
+  /**
+   * Gets the first available normal sample
+   * @returns the first available sample - uses 0-based array
+   */
+  public async getFirstAvailableNormalSample(): Promise<Locator> {
+    const samples = await this.getAllNormalSamples();
+    const amountOfSamples = samples.length;
+    expect.soft(amountOfSamples, 'There are no available normal samples in the Sequencing Tab').toBeGreaterThanOrEqual(1);
```

Comment:
should not use `expect.soft` here b/c `samples[0]` will throw error if array is empty.